include/xtensor/xfunctor_view.hpp

```diff
@@ -333,14 +333,26 @@ namespace xt
 
         bool equal(const xfunctor_iterator& rhs) const;
 
+        xfunctor_iterator operator+(difference_type offset)
+        {
+            return xfunctor_iterator(m_it + offset, p_functor);
+        }
+
+        difference_type operator-(xfunctor_iterator it2)
+        {
+            return m_it - it2.m_it;
+        }
+
+        bool operator<(xfunctor_iterator it2)
+        {
+            return (*this - it2) < 0;
+        }
+
```

Comment:
Indeed, it seems that there was clearly an issue with `operator+`, which should take a `difference_type`.

Is it necessary to have these be methods?

Reply:
Quick formatting note: we separate the implementation from the declaration even in templates in all the xtensor code base.

Reply:
For some reason VC14 kept giving me an ambigous match on all my attempts to use a friend function. Will separate the body.